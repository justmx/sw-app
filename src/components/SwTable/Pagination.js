import React from "react";

const Pagination = ({ nextPage, previouPage }) => {
  return (
    <div className="pagination">
      <span className="pagination-title">Pagination</span>
      <span className="pagination-button" onClick={previouPage}>
        {"< Back"}
      </span>
      <span className="pagination-button" onClick={nextPage}>
        {"Next >"}
      </span>
    </div>
  );
};

export default Pagination;
