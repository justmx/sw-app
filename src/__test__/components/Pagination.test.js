import React from "react";
import { shallow } from "enzyme";
import Pagination from "../../components/SwTable/Pagination";

const nextPage = jest.fn();
const previouPage = jest.fn();

describe("RowPaginationItem", () => {
  it("should render row itme", () => {
    const wrapper = shallow(
      <Pagination nextPage={jest.fn()} previouPage={jest.fn()} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should fire a function when click buttons", () => {
    const wrapper = shallow(
      <Pagination nextPage={nextPage} previouPage={previouPage} />
    );
    wrapper.find(".pagination-button").at(0).simulate("click");
    expect(previouPage).toHaveBeenCalled();
    wrapper.find(".pagination-button").at(1).simulate("click");
    expect(nextPage).toHaveBeenCalled();
  });
});
