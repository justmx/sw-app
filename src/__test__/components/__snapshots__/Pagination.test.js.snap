// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowPaginationItem should render row itme 1`] = `
<div
  className="pagination"
>
  <span
    className="pagination-title"
  >
    Pagination
  </span>
  <span
    className="pagination-button"
    onClick={[MockFunction]}
  >
    &lt; Back
  </span>
  <span
    className="pagination-button"
    onClick={[MockFunction]}
  >
    Next &gt;
  </span>
</div>
`;
